modules/ipc/src/zenoh/session.cpp

```diff
@@ -3,17 +3,52 @@
 //=================================================================================================
 #include "hephaestus/ipc/zenoh/session.h"
 
+#include <algorithm>
+#include <cstddef>
 #include <memory>
+#include <string>
 #include <utility>
 
 #include <fmt/format.h>
 #include <zenoh/api/config.hxx>
 #include <zenoh/api/session.hxx>
 
+#include "hephaestus/telemetry/log.h"
+#include "hephaestus/telemetry/log_sink.h"
 #include "hephaestus/utils/exception.h"
+#include "hephaestus/utils/string/string_utils.h"
+#include "hephaestus/utils/utils.h"
 
 namespace heph::ipc::zenoh {
 namespace {
+
+[[nodiscard]] auto createSessionId(const std::string& id) -> std::string {
+  static constexpr std::size_t MAX_SESSION_ID_SIZE = 16;
+
+  throwExceptionIf<InvalidParameterException>(
+      !utils::string::isAlphanumericString(id),
+      fmt::format("invalid session id: {}, must be an alphanumeric string", id));
+
+  auto session_id = id;
+  if (session_id.size() > MAX_SESSION_ID_SIZE) {
+    heph::log(heph::WARN, "session id is too long, truncating", "session_id", session_id, "max_size",
+              MAX_SESSION_ID_SIZE);
+    session_id = session_id.substr(0, MAX_SESSION_ID_SIZE);
+  }
+
+  // Zenoh reverse this.
+  std::ranges::reverse(session_id);
+  return utils::string::toAsciiHex(session_id);
+}
+
+[[nodiscard]] auto createSessionIdFromBinaryName() -> std::string {
```

Comment:
Just thinking out loud: What happens if zenoh session names collide? If sb has a configurable node that might exist multiple times in a system but with different params (e.g. maybe a mcap recorder node running multiple times for different purposes). Would adding the PID to the binary name make sense? (like `97566_bag_recorder`)

Reply:
good point! Added